src/HotChocolate/Core/src/Types/Types/Scalars/JsonType.cs

```diff
@@ -181,7 +181,7 @@ public static JsonElement Format(IValueNode node)
             using var jsonWriter = new Utf8JsonWriter(bufferWriter);
             _visitor.Visit(node, new JsonFormatterContext(jsonWriter));
 
```

Comment:
jsonWriter.Flush?

Reply:
If you flush or close the writer, it advances the underlying buffer past the content

Reply:
Not following, the buffer is the internal memory buffer we use to write the json to. we can flush it an read it  then dispose it .... we anyway dispose it.

Reply:
@michaelstaib It's difficult to explain, but I'll try to walk you through it.

My initial thought was that the buffer needed to be flushed, and that would solve it, but it turns out that was wrong. I will go through line-by-line a variation I tested thoroughly:

```C#
using var bufferWriter = new ArrayWriter();
using (var jsonWriter = new Utf8JsonWriter(bufferWriter))  // Using here closes and flushes the writer after visitor
    _visitor.Visit(node, new JsonFormatterContext(jsonWriter));

var jsonReader = new Utf8JsonReader(bufferWriter.GetSpan(bufferWriter.Length)); // The length here is now 2, however by disposing the jsonWriter, the `_start` variable on the bufferWriter is also 2 (since `Advance` was called), so the only remaining data in the buffer is all `0`'s.
return JsonElement.ParseValue(ref jsonReader);
```

![image](https://user-images.githubusercontent.com/969938/230393183-c1e0bf03-3ecc-4e32-b2e2-b2d8afbbace8.png)

Calling just .Flush has the same effect.

Reply:
@michaelstaib here's an alternative:

![image](https://user-images.githubusercontent.com/969938/230399243-cd97b3f6-a81f-46d7-b6d9-9c1453270458.png)

Reply:
The analysis is wrong :) 

There was a bug ... and that was that I used the GetSpan(). 

GetSpan is a writer method to write into a segment of the writer ... in order to access the context you need to use `Body`. I fixed that and updated the PR. 

Body access the written data....

Your fix worked for this example but would have failed for larger jsons as you could just capture the pending bytes ... but when the writer does autoflush inbetween it would have been broken.

Reply:
We still need some proper tests for this here: `JsonTypeTests`

Reply:
> Your fix worked for this example but would have failed for larger jsons as you could just capture the pending bytes ... but when the writer does autoflush inbetween it would have been broken.

Ahh yes, good point. I think your fix looks good though.